base/sparse/sparsematrix.jl

```diff
@@ -3007,3 +3009,28 @@ end
 (+)(A::SparseMatrixCSC, J::UniformScaling) = A + J.λ * speye(A)
 (-)(A::SparseMatrixCSC, J::UniformScaling) = A - J.λ * speye(A)
 (-)(J::UniformScaling, A::SparseMatrixCSC) = J.λ * speye(A) - A
+
+#set-like operations
+function union!{Tv,Ti<:Integer}(s::Set,x::SparseMatrixCSC{Tv,Ti})
+    vals = nonzeros(x)
+    s = union!(s,vals)
+    if length(x) == length(vals)
+        push!(s,zero(Tv))
+    end
+    s
+end
+function unique{Tv,Ti<:Integer}(x::SparseMatrixCSC{Tv,Ti})
+    #assumes non zero values stored in correct column major order
+    vals = nonzeros(x)
+    if length(x) == length(vals)
+        return unique(vals)
+    else
+        vs = Tv[x[1]]
+        i = Ti(2)
+        while vs[end] != zero(Tv) && i <= length(vals)
```

Comment:
More profoundly, I don't understand what you're doing here. Why not simply do the equivalent of `unique([zero(Tv); vals])`?